src/print.rs

```diff
@@ -493,21 +493,18 @@ fn preset_list() -> String {
 #[cfg(test)]
 mod test {
     use super::*;
-    use crate::config::StarshipConfig;
     use crate::test::default_context;
     use crate::utils;
 
     #[test]
     fn main_prompt() {
-        let mut context = default_context();
-        context.config = StarshipConfig {
-            config: Some(toml::toml! {
+        let mut context = default_context().set_config(toml::toml! {
                 add_newline=false
                 format="$character"
                 [character]
                 format=">\n>"
-            }),
-        };
+        });
+
         context.root_config.format = "$character".to_string();
         context.target = Target::Main;
         context.root_config.add_newline = false;
```

Comment:
`set_config()` can handle `root_config` now here and elsewhere:
```suggestion
        context.target = Target::Main;
```